core/src/main/java/org/elasticsearch/search/suggest/SuggestBuilder.java

```diff
@@ -212,5 +227,68 @@ public T shardSize(Integer shardSize) {
             this.shardSize = shardSize;
             return (T)this;
         }
+
+        @Override
+        public String getWriteableName() {
+            return name;
+        }
+
+        @Override
+        public final T readFrom(StreamInput in) throws IOException {
+            T builder = doReadFrom(in);
+            builder.setName(in.readString());
+            builder.setSuggester(in.readString());
+            builder.text(in.readOptionalString());
+            builder.setPrefix(in.readOptionalString());
+            builder.setRegex(in.readOptionalString());
+            builder.field(in.readOptionalString())
+                   .analyzer(in.readOptionalString())
+                   .size(in.readOptionalVInt())
+                   .shardSize(in.readOptionalVInt());
+            return builder;
+        }
+
+        @Override
+        public final void writeTo(StreamOutput out) throws IOException {
+            doWriteTo(out);
```

Comment:
Okay, I begin to see why this might be a bit weird now. The sub-type will need to write the name first, because it needs it to construct the TermSuggestionBuilder. This is a bit odd, I don't think we had this case before, maybe @colings86 has an idea how to sove this more elegantly?